File: 백준/Bronze/2455. 지능형 기차/지능형 기차.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const answer = (input) => {
    let ret = 0, curr = 0;
    
    input.map((e) => {
        const [o, i] = e.split(' ').map(e => Number(e));
        curr += i - o;
        ret = Math.max(ret, curr);
    })

    return ret;
}

console.log(answer(input));

File: 백준/Bronze/2455. 지능형 기차/지능형 기차.cc
#include <bits/stdc++.h>
using namespace std;

int out, in, curr, mx;

int main() {
  for (int i = 0; i < 4; i++) {
    cin >> out >> in;
    curr += in - out;
    mx = max(mx, curr);
  }

  cout << mx;
  return 0;
}